File: android/app/src/main/kotlin/com/wagmilabs/sigil/ui/implementorinit/ProgressViews.kt
package com.wagmilabs.sigil.ui.implementorinit

import androidx.compose.foundation.layout.*
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.AccessTime
import androidx.compose.material.icons.filled.CheckCircle
import androidx.compose.material3.*
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp

@Composable
fun GeneratingView() {
    Column(
        horizontalAlignment = Alignment.CenterHorizontally,
        verticalArrangement = Arrangement.spacedBy(16.dp),
        modifier = Modifier.padding(24.dp)
    ) {
        CircularProgressIndicator()
        Text("Generating Mnemonic...", style = MaterialTheme.typography.titleMedium)
        Text(
            "Creating a secure 24-word recovery phrase",
            color = MaterialTheme.colorScheme.onSurfaceVariant
        )
    }
}

@Composable
fun EncryptingSubmittingView() {
    Column(
        horizontalAlignment = Alignment.CenterHorizontally,
        verticalArrangement = Arrangement.spacedBy(16.dp),
        modifier = Modifier.padding(24.dp)
    ) {
        CircularProgressIndicator()
        Text("Encrypting & Submitting...", style = MaterialTheme.typography.titleMedium)
        Text(
            "Securing mnemonic for implementor",
            color = MaterialTheme.colorScheme.onSurfaceVariant
        )
    }
}

@Composable
fun AwaitingConfirmationView() {
    Column(
        horizontalAlignment = Alignment.CenterHorizontally,
        verticalArrangement = Arrangement.spacedBy(16.dp),
        modifier = Modifier.padding(24.dp)
    ) {
        Icon(
            imageVector = Icons.Default.CheckCircle,
            contentDescription = null,
            tint = Color.Green,
            modifier = Modifier.size(64.dp)
        )
        Text("Submitted", style = MaterialTheme.typography.titleLarge)
        Text(
            "Waiting for implementor to confirm receipt...",
            textAlign = TextAlign.Center,
            color = MaterialTheme.colorScheme.onSurfaceVariant
        )
        CircularProgressIndicator()
    }
}

@Composable
fun ClaimingView() {
    Column(
        horizontalAlignment = Alignment.CenterHorizontally,
        verticalArrangement = Arrangement.spacedBy(16.dp),
        modifier = Modifier.padding(24.dp)
    ) {
        CircularProgressIndicator()
        Text("Claiming Request...", style = MaterialTheme.typography.titleMedium)
        Text(
            "Verifying claim code",
            color = MaterialTheme.colorScheme.onSurfaceVariant
        )
    }
}

@Composable
fun AwaitingPushView(timeRemaining: Int) {
    Column(
        horizontalAlignment = Alignment.CenterHorizontally,
        verticalArrangement = Arrangement.spacedBy(16.dp),
        modifier = Modifier.padding(24.dp)
    ) {
        CircularProgressIndicator()
        Text("Waiting for Request...", style = MaterialTheme.typography.titleMedium)
        Text(
            "Your claim was successful. Waiting for the mnemonic generation request.",
            textAlign = TextAlign.Center,
            color = MaterialTheme.colorScheme.onSurfaceVariant
        )

        if (timeRemaining < 30) {
            Row(horizontalArrangement = Arrangement.spacedBy(8.dp)) {
                Icon(Icons.Default.AccessTime, contentDescription = null)
                Text("Timeout in ${timeRemaining}s")
            }
        }
    }
}

File: ios/Sources/UI/ImplementorInit/ProgressViews.swift
import SwiftUI

// Generating progress
public struct GeneratingView: View {
    public init() {}

    public var body: some View {
        VStack(spacing: 24) {
            ProgressView()
                .controlSize(.large)

            Text("Generating Mnemonic...")
                .font(.title3)
                .fontWeight(.medium)

            Text("Creating a secure 24-word recovery phrase")
                .font(.body)
                .foregroundColor(.secondary)
        }
        .padding()
    }
}

// Encrypting & Submitting
public struct EncryptingSubmittingView: View {
    public init() {}

    public var body: some View {
        VStack(spacing: 24) {
            ProgressView()
                .controlSize(.large)

            Text("Encrypting & Submitting...")
                .font(.title3)
                .fontWeight(.medium)

            Text("Securing mnemonic for implementor")
                .font(.body)
                .foregroundColor(.secondary)
        }
        .padding()
    }
}

// Awaiting Confirmation
public struct AwaitingConfirmationView: View {
    public init() {}

    public var body: some View {
        VStack(spacing: 24) {
            Image(systemName: "checkmark.circle.fill")
                .font(.system(size: 64))
                .foregroundColor(.green)

            Text("Submitted")
                .font(.title2)
                .fontWeight(.semibold)

            Text("Waiting for implementor to confirm receipt...")
                .font(.body)
                .foregroundColor(.secondary)
                .multilineTextAlignment(.center)

            ProgressView()
        }
        .padding()
    }
}

// Claiming
public struct ClaimingView: View {
    public init() {}

    public var body: some View {
        VStack(spacing: 24) {
            ProgressView()
                .controlSize(.large)

            Text("Claiming Request...")
                .font(.title3)
                .fontWeight(.medium)

            Text("Verifying claim code")
                .font(.body)
                .foregroundColor(.secondary)
        }
        .padding()
    }
}

// Awaiting Push (after successful claim)
public struct AwaitingPushView: View {
    let timeRemaining: Int

    public init(timeRemaining: Int) {
        self.timeRemaining = timeRemaining
    }

    public var body: some View {
        VStack(spacing: 24) {
            ProgressView()
                .controlSize(.large)

            Text("Waiting for Request...")
                .font(.title3)
                .fontWeight(.medium)

            Text("Your claim was successful. Waiting for the mnemonic generation request.")
                .font(.body)
                .foregroundColor(.secondary)
                .multilineTextAlignment(.center)

            if timeRemaining < 30 {
                HStack {
                    Image(systemName: "clock")
                    Text("Timeout in \(timeRemaining)s")
                }
                .font(.caption)
                .foregroundColor(.secondary)
            }
        }
        .padding()
    }
}
